Edit readme to include forced overrides functionability (#2233)
SalesforceCommerceCloud/pwa-kit: packages/pwa-kit-extension-sdk/README.md

This mechanism is useful when you want to allow for fine grained customization of your application extension. Its the responsibility of the extension developer to ensure their document what files are overridable and what the expected input and output of those files are. For example, if the overridable file is a React component you should document the props that get passed to that component and the expected exports of the file.

> #### Forced overrides

Reviewer: To match how other headings are capitalized:
```suggestion
#### Forced Overrides
```

> Sometimes, you may need to override some files from an extension but the extension developers do not allow those files to be overridden, you can use `forced_overrides` to bypass these restrictions.

Reviewer: 🤔 I'm not sure whether this is the right place to describe forced overrides. Because at this point, the readme is talking about `The anatomy of an Application Extension` (see line 25). But forced overrides is a feature that is used in the base project (not in the app extension).

So what is the correct place? I'm thinking `template-typescript-minimal/README.md` file. But that readme does not have any existing info on app extensibility.

Author: Typescript minimal is a private project atm. I asked the same question previously too, Adam advised we can put it here to explain about overrides machenism since the logic stays in this package. I think it is okay since we talked about overrides in this section. it is logical we can mentioned about force_overrides too

Reviewer: Even though typescript-minimal is a private package, its readme will be part of the generated project. So other folks will still be able to read it in their project.

Reviewer: If we still want to mention forced overrides in this file, let's rewrite it a bit, so that it makes more sense to the audience in this case, which are the extension developers. Perhaps something like this:

> There may be cases where the users of your extension may want to override some of your files that you have not allowed to be extended. In this case, please know that there exists `forced_overrides` for them to use to bypass these restrictions...

> In a case when it is necessary to override some files from an extension but the extension developers do not allow those files to be overridden, you can use `forced_overrides` to bypass the restrictions.

Reviewer 2: optional nit: rephrasing a bit
```suggestion
Sometimes, you may need to override files from an extension but the extension developers do not allow those files to be overridden, you can use `forced_overrides` to bypass these restrictions.
```

> To do this, create a `.forced_overrides` dot file at the root of your base project and list the files you intend to override. Any files specified in this list will be treated as overridable.

Reviewer 2: optional nit: expand on how to specify the files
 ```suggestion
To do this, create a `.forced_overrides` dot file at the root of your base project, list the files you intend to override, starting with the extension package name. Any files specified in this list will be treated as overridable.
```

> Note: It is advisable to use this function with caution since it could potential break intended behavior of installed extensions.   

Reviewer 2: I'd also note that this should used as a temporary workaround while seeking a proper extensibility solution. Like we mention in the file docs.
https://github.com/SalesforceCommerceCloud/pwa-kit/blob/1f2bc449f04c1b102889607310c31eb28d1ab330/packages/template-typescript-minimal/.force_overrides#L5

> Note: It is advisable to use this function with caution since it could potential break intended behavior of installed extensions. This should be used as a temporary solution to urgently patch/update an extension.   

Reviewer: ```suggestion
Note: It is advisable to use this function with caution since it could potentially break intended behavior of installed extensions. This should be used as a temporary solution to urgently patch/update an extension.   
```
